src/api.rs

```diff
@@ -1685,6 +1666,72 @@ impl<'a> AuthenticatedApi<'a> {
             resp.convert()
         }
     }
+
+    pub fn region_specific<'b>(&'a self, org: &'b str) -> RegionSpecificApi<'a, 'b> {
+        let base_url = self.api.config.get_base_url();
+        if base_url.is_err()
+            || base_url.expect("base_url should not be error") != DEFAULT_URL.trim_end_matches('/')
+        {
+            // Do not specify a region URL unless the URL is configured to https://sentry.io (i.e. the default).
+            return RegionSpecificApi {
+                api: self,
+                org,
+                region_url: None,
+            };
+        }
+
+        match self
+            .api
+            .config
+            .get_auth()
+            .expect("auth should not be None for authenticated API!")
+        {
+            Auth::Token(token) if token.payload().is_some() => {
+                let region_url = &token
+                    .payload()
+                    .expect("Payload already checked to have Some value")
+                    .region_url;
+
+                RegionSpecificApi {
+                    api: self,
+                    org,
+                    region_url: Some(region_url),
+                }
+            }
+            _ => {
+                log::info!(
+                    "Auth does not encode a region URL. Falling back to using default region!"
+                );
+                RegionSpecificApi {
+                    api: self,
+                    org,
+                    region_url: None,
+                }
+            }
+        }
+    }
+}
+
+impl<'a, 'b> RegionSpecificApi<'a, 'b> {
+    fn request(&self, method: Method, url: &str) -> ApiResult<ApiRequest> {
```

Comment:
I wonder if we can eventually extract out these methods as a general API trait, providing a regionless implementation as a fallback? It might be a ton of extra plumbing though for the fallback cases 😬